### ISSUE TITLE
Backtrace detection is broken on JRuby 9000

### ISSUE THREAD

**Issue by JonRowe** [rspec/rspec-core#2299]
JRubys 9000's (specifically 9.1.2.0) backtraces are no longer being detected properly resulting in "unable to find backtrace line" for several of our specs and two features in our build.

It would seem these are all related to detecting backtraces from outside a users project code (external libraries and our pending exceptions).
- [x] spec/integration/failed_line_detection_spec.rb:42
- [x] spec/integration/failed_line_detection_spec.rb:104
- [x] spec/integration/failed_line_detection_spec.rb:138
- [ ] spec/rspec/core/example_group_spec.rb:1033
- [ ] spec/rspec/core/example_spec.rb:649
- [ ] features/configuration/backtrace_exclusion_patterns.feature:127
- [ ] features/expectation_framework_integration/configure_expectation_framework.feature:76

**Comment by xaviershay** [comment]
Quick thing I just checked: current jruby 9000 build "passes" on Travis but actually it's not running the specs: https://travis-ci.org/rspec/rspec-core/jobs/192034497

**Comment by myronmarston** [comment]
> Quick thing I just checked: current jruby 9000 build "passes" on Travis but actually it's not running the specs: https://travis-ci.org/rspec/rspec-core/jobs/192034497

Sure looks like it's running the specs to me.  Did you try expanding the collapsed output sections?  The one labeled "specs" (the command is `/home/travis/build/rspec/rspec-core/bin/rspec`) shows the RSpec output.

**Comment by DavidEGrayson** [comment]
This is affecting me too.  I am getting this in my RSpec 3.5 output on JRuby 9.1.7.0 on Windows:

    Failure/Error: Unable to find matching line from backtrace

I traced through the RSpec code, going backwards in time from that error message and I determined that [line 267 of example.rb in rspec-core v3.5.4](https://github.com/rspec/rspec-core/blob/v3.5.4/lib/rspec/core/example.rb#L267). is rescuing an exception that has no backtrace for some reason.
